api/src/main/resources/workbench.yaml

```diff
@@ -960,6 +960,54 @@ paths:
           description: A list of resources order by last access time desc order
           schema:
             $ref: "#/definitions/RecentResourceResponse"
+
+  /v1/recentResource/{workspaceId}/{workspaceName}/{notebook}/delete-notebook:
+    parameters:
+      - $ref: '#/parameters/workspaceNamespace'
+      - $ref: '#/parameters/workspaceId'
+    delete:
+      tags:
+        - userMetrics
+      description: Deletes Notebook entry from recent resource
+      operationId: deleteNotebookEntry
+      parameters:
+        - in: path
+          name: notebook
+          required: true
+          type: string
+      responses:
+        200:
+          description: Successfully deleted
+          schema:
+            $ref: "#/definitions/EmptyResponse"
+
+  /v1/recentResource/{workspaceId}/{workspaceName}/rename-notebook:
```

Comment:
This feels to me like it might be overly tuned to the use case. I would recommend just providing a delete and update method, I think the Jupyter server extension can make do with just that.